src/Sarif.Converters/PREFastConverter.cs

```diff
@@ -0,0 +1,193 @@
+﻿using System.IO;
+using Microsoft.CodeAnalysis.Sarif.Writers;
+using System;
+using System.Collections.Generic;
+using System.Xml;
+using System.Xml.Serialization;
+using Microsoft.CodeAnalysis.Sarif.Converters.PREFastObjectModel;
+using System.Linq;
+
+namespace Microsoft.CodeAnalysis.Sarif.Converters
+{
+    public class PREfastConverter : ToolFileConverterBase
+    {
+        private Dictionary<string, string> categories = new Dictionary<string, string>
+        {
+            { "RULECATEGORY", "ruleCategory" }
+        };
+
+        public override void Convert(Stream input, IResultLogWriter output, LoggingOptions loggingOptions)
+        {
+            input = input ?? throw new ArgumentNullException(nameof(input));
+            output = output ?? throw new ArgumentNullException(nameof(output));
+            output.Initialize(id: null, automationId: null);
+
+            var tool = new Tool
+            {
+                Name = ToolFormat.PREfast,
+                FullName = "PREfast Code Analysis"
+            };
+
+            output.WriteTool(tool);
+
+            var serializer = new XmlSerializer(typeof(DefectList));
+            var defectList = (DefectList)serializer.Deserialize(input);
+            var results = new List<Result>();
+            foreach (Defect entry in defectList.Defects)
+            {
+                results.Add(CreateResult(entry));
+            }
+
+            var fileInfoFactory = new FileInfoFactory(MimeType.DetermineFromFileExtension, loggingOptions);
+            Dictionary<string, FileData> fileDictionary = fileInfoFactory.Create(results);
+
+            if (fileDictionary?.Any() == true)
+            {
+                output.WriteFiles(fileDictionary);
+            }
+
+            output.OpenResults();
+            output.WriteResults(results);
+            output.CloseResults();
+        }
+
+        private Result CreateResult(Defect defect)
+        {
+            var region = new Region
+            {
+                StartColumn = defect.SFA.Column + 1,
+                StartLine = defect.SFA.Line
+            };
+
+            Uri resultsFileUri = new Uri($"{defect.SFA.FilePath}{defect.SFA.FileName}", UriKind.Relative);
+            PhysicalLocation physicalLocation = new PhysicalLocation(uri: resultsFileUri, uriBaseId: null, region: region);
+            Location location = new Location()
+            {
```

Comment:
You can use `var` for these three variables since the types are all explicit on the right-hand side.

Reply:
fixed

---
In reply to: [146949725](https://github.com/Microsoft/sarif-sdk/pull/657#discussion_r146949725) [](ancestors = 146949725)